example/calculate_pi/calculate_pi.cpp

```diff
@@ -71,12 +71,29 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   Metadata m({Metadata::Cell, Metadata::Derived});
   package->AddField(field_name, m, DerivedOwnership::unique);
 
+  // Add a named MeshPack by registering a function that packs it
+  package->AddMeshBlockPack("in_or_out", [](Mesh *pmesh) {
+    // 1 means pack is 1 meshblock, <1 means use entire mesh
+    int pack_size = pmesh->DefaultPackSize();
+    if (pack_size < 1) pack_size = pmesh->block_list.size();
+    std::vector<BlockList_t> partitions;
+    std::vector<MeshBlockVarPack<Real>> packs;
+    partition::ToSizeN(pmesh->block_list, pack_size, partitions);
+    packs.resize(partitions.size());
```

Comment:
Suggestion: `partition::ToSizeN` could just take the `pmesh->block_list` and `pack_size` args and return the `std::vector<BlockList_t> partitions`.  Then you'd do something like
```c++
std::vector<MeshBlockVarPack<Real>> packs(partitions.size());
```

Reply:
The reason I didn't go that route is I didn't want to return a `std::vector` for performance reasons. But if we wrap it in a `std::move` it should be fine.

Reply:
I don't think the `std::move` is actually required in this case.

Reply:
Well you'd need a line like
```C++
std::vector<MeshBlockVarPack<Real>> packs(partition::ToSizeN(pmesh->block_list,pack_size));
```
right? And this *might* automatically do a `std::move` but you could enforce it with 
```C++
std::vector<MeshBlockVarPack<Real>> packs = std::move(partition::ToSizeN(pmesh->block_list,pack_size));
```
Or am I missing something?

Reply:
Not sure I follow.  I was suggesting something like
```c++
auto partitions = partition::ToSizeN(pmesh->block_list, pack_size);
std::vector<MeshBlockVarPack<Real>> packs(partitions.size());
```
I believe if you make `partition::ToSizeN` just return the local-to-that-function `std::vector<BlockList_t>` the compiler will take care of the rest.  In fact, I believe this is the preferred way to write such a capability.  No explicit `std::move`s should be used.  I could be wrong, of course.

Reply:
I might be stuck in the past of C++. I remember the advice not to return things explicitly like that. Either way, easy to check. Just switch the design and see what happens.

Also if it's only called once per remeshing, the difference is almost certainly negligible.

Reply:
Yeah I'm not really worried about performance.  I just think it's cleaner this way.

Reply:
Agreed. I'll make that change.

Reply:
Getting type inference to work with this change required a little template foo, but I got it.